crates/agglayer/src/main.rs

```diff
@@ -31,7 +31,18 @@ fn main() -> anyhow::Result<()> {
             let vkey = agglayer_prover::get_vkey();
             println!("{}", vkey);
         }
+        cli::Commands::Version => {
+            println!("version info: {}", version());
+        }
     }
 
     Ok(())
 }
+
+/// Common version information about the executed agglayer binary.
+pub fn version() -> String {
+    let pkg_name = env!("CARGO_PKG_NAME");
+    let git_describe = env!("VERGEN_GIT_DESCRIBE");
+    let timestamp = env!("VERGEN_BUILD_TIMESTAMP");
```

Comment:
How is the timestamp obtained? It seems a bit odd if just running `cargo build` some time later with no other changes results in a different binary. Imho that sort of non-determinism should be limited. If it actually takes the current time, i's advise against it. On the other hand, taking the timestamp from the latest commit is fine.

Reply:
It takes the timestamp from the local system in the moment of the build. You can uniquely identify the commit with the commit hash available with the `VERGEN_GIT_DESCRIBE` and that does not change in the other build of the same commit.
I don't think the time of the build hurts, it is clearly marked as the build time information - nothing more, nothing less.

Reply:
Not to digress too far, even `Git Release process` has its own date/time when the binaries/artifacts are created/packed and uploaded on the release page, not the time of the latest commit that was released.

Reply:
> It takes the timestamp from the local system in the moment of the build. You can uniquely identify the commit with the commit hash available with the `VERGEN_GIT_DESCRIBE` and that does not change in the other build of the same commit. I don't think the time of the build hurts, it is clearly marked as the build time information - nothing more, nothing less.

There are some drawbacks in builds not being deterministic. For example, it makes it more difficult for third parties to check the result of the build by reproducing it with the same build inputs. Consider building a Docker image with agglayer node in it. It can uniquely identified by a hash. However, we will get different resulting image hash even if the exact same sources are used to build the node.

More info [here](https://reproducible-builds.org/). Vergen has a section on deterministic outputs [here](https://docs.rs/vergen/9.0.2/vergen/#support-deterministic-output).

Reply:
> Having the very same agglayer source commit built on the two different machines - you will certainly have different agglayer file hash - and docker image hash built at different time with the same commit will certainly be different. It will depend on many different parameters - file system creation times, compiler version etc. A little experiment - take the current `origin/main`, build the `agglayer` binary and check the `md5sum` of it. Then `cargo clean`, change the compiler version and repeat the same. They will have different md5 hash.

Yea, the output depends on many variables, including the directory path where the build happens etc, so it's not very practical at the moment. However, I'd say it's preferable if we directionally move towards more determinism, or at least not introduce much more of it.